libs/vkd3d/device.c

```diff
@@ -9517,6 +9526,66 @@ bool d3d12_device_validate_shader_meta(struct d3d12_device *device, const struct
         }
     }
 
+    if (meta->flags & VKD3D_SHADER_META_FLAG_USES_COOPERATIVE_MATRIX)
+    {
+        const struct vkd3d_vk_instance_procs *vk_procs = &device->vkd3d_instance->vk_procs;
+        VkCooperativeMatrixPropertiesKHR *props;
+        bool supports_f32_16x16x16_f16 = false;
+        bool supports_f16_16x16x16_f16 = false;
+        bool supports_u8_a = false;
+        bool supports_u8_b = false;
+        uint32_t i, count;
+
+        /* There are no sub-capabilities (yet at least).
+         * Validate that we support everything that dxil-spirv can emit. */
+        if (VK_CALL(vkGetPhysicalDeviceCooperativeMatrixPropertiesKHR(
+                device->vk_physical_device, &count, NULL)) != VK_SUCCESS)
```

Comment:
does it make sense to cache this at a device level or do we expect queries to change depending on the shader in the future?

Reply:
It could make sense to cache it.